### PR TITLE
Use turbo ordinals wallet

### PR DESCRIPTION
**Pull request by dannydeezy**

### CODE REVIEW

**Comment by dannydeezy** [comment]
Screenshot 2023-03-14 at 1 16 05 PM
for some reason the images are all the same here on the deploy preview, but when i checkout and run locally they are correct

**Comment by dannydeezy** [comment]
Screenshot 2023-03-14 at 4 19 24 PM
we need to do the same for this page

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
         "react-social-icons": "^5.15.0",
         "react-toastify": "^9.1.1",
         "react-useinterval": "^1.0.2",
+        "redis": "^4.6.5",
```
Comment (habibitcoin) [inline package.json:42]: nit: can remove?

**Review by habibitcoin — APPROVED** [review]
Working great, and tried out with some non-image ordinals as well and no issues with scripts running maliciously [either](https://ordinalswallet.com/inscription/13c44582fcaaaa717cef8956d7548ccf25099a6a2d2e6e7684746721726732e1i0) 


Nice work! Much faster

**Comment by andredevjs** [comment]
@dannydeezy @habibitcoin ready to go?

File: src/containers/OrdinalsArea.js [hunk 2]
```diff
@@ -35,7 +36,8 @@ const getSortedUtxos = async (nostrAddress) => {
 };
 
 const OrdinalsArea = ({ className, space }) => {
-    const { nostrAddress } = useContext(WalletContext);
+    // const { nostrAddress } = useContext(WalletContext);
+    const nostrAddress = "bc1p8l0pstx8umh6dx3e8vtw7sd3pspe9r0nh94v7ncwkqleljnr5zdqa3cvlm";
```
Comment (habibitcoin) [inline src/containers/OrdinalsArea.js:40]: Oops

File: src/components/inscription-preview/index.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+export const InscriptionPreview = ({ utxo }) => {
+    const [loading, setLoading] = useState(true);
+
+    const isImage = /(^image)(\/)[a-zA-Z0-9_]*/gm.test(utxo.content_type);
+    const handleLoad = () => {
+        setLoading(false);
+    };
+
+    const renderImage = () => {
```
Comment (habibitcoin) [inline src/components/inscription-preview/index.js:14]: love ittt

File: src/components/modals/send-modal/index.js [hunk 4]
```diff
@@ -102,14 +102,9 @@ const SendModal = ({ show, handleModal, utxo, onSale }) => {
...
-                <IframeWithLoader
-                    id="preview"
-                    sandbox="allow-scripts allow-same-origin"
-                    scrolling="no"
-                    loading="lazy"
-                    title={utxo.inscriptionId}
-                    src={`${ORDINALS_EXPLORER_URL}/preview/${utxo.inscriptionId}`}
-                />
+                <div className="inscription-preview">
+                    <InscriptionPreview utxo={utxo} />
```
Comment (habibitcoin) [inline src/components/modals/send-modal/index.js:106]: nice

File: src/containers/OrdinalsArea.js [hunk 5]
```diff
@@ -72,19 +51,76 @@ const OrdinalsArea = ({ className, space }) => {
...
-            );
-            setOwnedUtxos(ownedInscriptionResults.map((utxo) => utxo.value));
+
+            const utxos = await getSortedUtxos(nostrAddress);
+            const inscriptions = await axios.get(`${TURBO_API}/wallet/${nostrAddress}/inscriptions`);
+
+            const inscriptionsByUtxoKey = {};
+            const batchPromises = [];
+            const populateInscriptionsMap = async (ins) => {
+                const outpoint = await getOutpointFromCache(ins.id);
```
Comment (habibitcoin) [inline src/containers/OrdinalsArea.js:82]: @andredevjs sorry 1 question, couldn't getOutpointFromCache be replaced with an await.fetch() and then utilize 'force-cache' mechanism to cache it server side? Then all users can benefit

Also maybe dumb question, but why do we sometimes use fetch and sometimes axios?

**Pull request merged**